rust/src/util/types.rs

```diff
@@ -0,0 +1,32 @@
+use std::ptr;
+
+use drop_struct_macro_derive::DropStructMacro;
+// `CodeAndMessage` is the trait implemented by `code_and_message_impl
+use ffi_toolkit::{code_and_message_impl, free_c_str, CodeAndMessage, FCPResponseStatus};
+
+#[repr(C)]
+#[derive(DropStructMacro)]
+pub struct GpuDeviceResponse {
+    pub status_code: FCPResponseStatus,
+    pub error_msg: *const libc::c_char,
+    pub devices_len: libc::size_t,
+    pub devices_ptr: *const *const i8,
```

Comment:
@vmx - NITPICK: We have used `*const libc::c_char` for C-strings elsewhere in the API. I see you've got a dynamic array of C-strings (`devices_ptr`) here, and that you used `*const i8` instead. I humbly request that you use `*const libc::c_char` instead of `*const i8`, for consistency.